metricbeat/mb/registry.go

```diff
@@ -46,6 +47,8 @@ type metricSetFactoryInfo struct {
 // Register contains the factory functions for creating new Modules and new
 // MetricSets.
 type Register struct {
+	//Lock to control concurrent read/writes
+	sync.RWMutex
```

Comment:
I think embedding the mutex is not a good choice here because the methods will be become part of `Register`'s public API. Users should never need to lock/unlock the `Register` because that is handled internally.

This reminds me, can you add to the godoc for `Register` that "It is safe for concurrent usage."`.